### DISCUSSION TITLE
Object control not follow with cursor

### DISCUSSION THREAD

**Discussion by CocaColf** [fabricjs/fabric.js#8702] [Q&A] [answered]
As the Gif shows, when I scale object by controls, it's just follow with the direction where cursor goes.

But if I want the object scale to the position of cursor, is there has any configuration?

![fabric](https://user-images.githubusercontent.com/25732253/219360850-90302e2d-11cd-443e-9978-2ad5e34bf47a.gif)

**Comment by iocoker** [comment, marked as answer]
If you hold down the shift key, that will scale non-proportionally and follow the mouse.

**Reply by CocaColf** [reply]
Oh thank you, it's really works! But I didn't find any information from the [docs](http://fabricjs.com/docs/fabric.Object.html) 

So if I want it don't need hold down shift key, is there any way to achieve it ? Thank you again!

**Reply by ShaMan123** [reply]
I didn't know that. Nice!
Looking at
https://github.com/fabricjs/fabric.js/blob/93dd2dcca705a4b481fbc9982da4952ef5b4ed1d/src/controls/scale.ts#L35-L45

Seems you need `canvas.uniformScaling = false`

**Reply by asturur** [reply]
uniformScaling is the default behaviour that can be toggled, and the shift key or  the one put under the `uniScaleKey` ( can be alt or control or cmd ) act as a modifier on top of the default behaviour

**Reply by CocaColf** [reply]
From the doc I can see `If null or none or any other string that is not a modifier key feature is disabled`.

So that's means I cannot make the scale behavior as uniformScaling by default? It's must combine with the [uniScaleKey](http://fabricjs.com/docs/fabric.Canvas.html#uniScaleKey) ?

Thank you and looking forward for your reply!

**Reply by iocoker** [reply]
You can make it default behavior by setting this: `canvas.uniformScaling = false`
